### PR TITLE
Fix XRandr for old systems and MacOSX

### PR DESCRIPTION
**Pull request by hardening**
We need a recent enough version of XRandr to correctly detect monitor.
Also this patch adds some typo or style fixes.

### CODE REVIEW

**Comment by hardening** [comment]
@diget @akallabeth I finally found what was wrong with MacOS X, but it was also affecting systems with old versions of XRandr

File: libfreerdp/core/rdp.c [hunk 1]
```diff
@@ -604,7 +604,7 @@ BOOL rdp_send_data_pdu(rdpRdp* rdp, wStream* s, BYTE type, UINT16 channel_id)
 	Stream_SetPosition(s, length);
 	Stream_SealLength(s);
 
-	WLog_DBG(TAG, "%s: sending data (type=0x%x size=%d channelId)", __FUNCTION__, type, Stream_Length(s), channel_id);
+	WLog_DBG(TAG, "%s: sending data (type=0x%x size=%d channelId=%d)", __FUNCTION__, type, Stream_Length(s), channel_id);
```
Comment (akallabeth) [inline libfreerdp/core/rdp.c]: Looks like there are a few issues here (not your fault, but none the less):
* ```Stream_GetPosition``` returns ```sizt_t```
  * Type of ```length``` is ```UINT16```
  * Type of ```sec_hold``` is ```int```
* Debug format string uses wrong format specifiers

Comment (hardening) [reply libfreerdp/core/rdp.c]: ok will change that

**Comment by hardening** [comment]
I took the opportunity to treat all callers of Stream_GetPosition()

File: libfreerdp/core/rdp.c [hunk 2]
```diff
@@ -604,7 +604,7 @@ BOOL rdp_send_data_pdu(rdpRdp* rdp, wStream* s, BYTE type, UINT16 channel_id)
 	Stream_SetPosition(s, length);
 	Stream_SealLength(s);
 
-	WLog_DBG(TAG, "%s: sending data (type=0x%x size=%d channelId=%d)", __FUNCTION__, type, Stream_Length(s), channel_id);
+	WLog_DBG(TAG, "%s: sending data (type=0x%x size=%d channelId=%"PRIu16")", __FUNCTION__, type, Stream_Length(s), channel_id);
```
Comment (akallabeth) [inline libfreerdp/core/rdp.c]: Wrong format string argument, size_t has its own defined ```PRIzd``` (or similar)

**Comment by hardening** [comment]
@akallabeth is it good for you now ?

**Comment by akallabeth** [comment]
@hardening Nice work ^^

**Pull request merged**